### PR TITLE
Drop `RequestStore` in favour of `ActiveSupport::CurrentAttributes`

### PR DESCRIPTION
**Pull request by joshuay03**
Thank you for your contribution!

Check the following boxes:

- [x] Wrote [good commit messages][1].
- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new 
  code introduces user-observable changes.
- [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://chris.beams.io/posts/git-commit/

Fixes https://github.com/paper-trail-gem/paper_trail/issues/1444

I wonder if there are any performance benefits here, I'm happy to run some benchmarks if it's a valid concern.

### CODE REVIEW

File: spec/controllers/widgets_controller_spec.rb [hunk 1]
```diff
@@ -5,8 +5,6 @@
 RSpec.describe WidgetsController, type: :controller, versioning: true do
   before { request.env["REMOTE_ADDR"] = "127.0.0.1" }
 
-  after { RequestStore.store[:paper_trail] = nil }
```
Comment (joshuay03) [inline spec/controllers/widgets_controller_spec.rb:8]: Current attributes are automatically reset before each request so this is no longer necessary for the new implementation.

**Comment by joshuay03** [comment]
CI is failing cause of reduced line coverage, but all my changes except for the new `CurrentAttributes` class (which I've added some coverage for) are already covered in existing tests...

**Comment by ngan** [comment]
@jaredbeck would you mind taking a look at this? 🙏

**Comment by achernik** [comment]
dear maintainers, please consider merging this PR

**Pull request closed without merging**